samples/I2CBMP280.js

```diff
@@ -138,38 +128,54 @@ function updateColor(newTemp) {
     }
 
     prevTemp = newTemp;
-    changeRGB(red, green, blue);
+
+    // Valid range for color is 0 - 255
+    redData.writeUInt8(red, 1);
+    greenData.writeUInt8(green, 1);
+    blueData.writeUInt8(blue, 1);
+    // Send messages as close together as possible,
+    // so that the color change is smoother
+    i2cDevice.write(glcd.GROVE_RGB_BACKLIGHT_ADDR, blueData);
+    i2cDevice.write(glcd.GROVE_RGB_BACKLIGHT_ADDR, redData);
+    i2cDevice.write(glcd.GROVE_RGB_BACKLIGHT_ADDR, greenData);
 }
 
 function writeWord(word, buffer) {
-    word = "@" + word;  //This is needed to write to the correct register address
-    buffer.write(word);
+    // Add the word to the buffer
+    buffer.write(word, 1);
+    // The first byte in the buffer is for the register address,
+    // which is where the word is going to be written to.
+    buffer.writeUInt8(glcd.GLCD_CMD_SET_CGRAM_ADDR, 0);
+    // Write the buffer to I2C
     i2cDevice.write(glcd.GROVE_LCD_DISPLAY_ADDR, buffer);
 }
 
 function init() {
-    reader = i2cDevice.read(bmp280.BMP280_ADDRESS, 1,
+    var reader = i2cDevice.read(bmp280.BMP280_ADDRESS, 1,
                             bmp280.BMP280_REGISTER_CHIPID);
 
+    // Check that we have a BMP280 on the I2C bus
     if (reader.toString('hex') !== "58")
         print("BMP280 not found! ChipId " + reader.toString('hex') !== "58");
 
     readCoefficients();
 
-    var powerData = new Buffer([bmp280.BMP280_REGISTER_CONTROL, 0]);
-    powerData.writeUInt8(bmp280.BMP280_PRESS_OVER | bmp280.BMP280_TEMP_OVER |
+    //var powerData = new Buffer([bmp280.BMP280_REGISTER_CONTROL, 0]);
+    setupData.writeUInt8(bmp280.BMP280_REGISTER_CONTROL, 0);
+    setupData.writeUInt8(bmp280.BMP280_PRESS_OVER | bmp280.BMP280_TEMP_OVER |
                          bmp280.BMP280_NORMAL_MODE, 1)
 
-    i2cDevice.write(bmp280.BMP280_ADDRESS, powerData);
+    i2cDevice.write(bmp280.BMP280_ADDRESS, setupData);
 
-    powerData.writeUInt8(bmp280.BMP280_REGISTER_CONFIG, 0);
-    powerData.writeUInt8(bmp280.BMP280_STANDBY | bmp280.BMP280_FILTER |
+    setupData.writeUInt8(bmp280.BMP280_REGISTER_CONFIG, 0);
+    setupData.writeUInt8(bmp280.BMP280_STANDBY | bmp280.BMP280_FILTER |
                          bmp280.BMP280_SPI_3W_DISABLE, 1);
 
-    i2cDevice.write(bmp280.BMP280_ADDRESS, powerData);
+    i2cDevice.write(bmp280.BMP280_ADDRESS, setupData);
 
     /* Setup Grove LCD */
```

Comment:
No, everything below that is part of setting up the LCD.  Clear is just the first line of  the code after that.